extensions/lifecycle/bucketProcessor/task.js

```diff
@@ -17,55 +23,46 @@ werelogs.configure({
 
 const logger = new werelogs.Logger('Backbeat:Lifecycle:Producer');
 
-const bucketProcessor = new LifecycleBucketProcessor(zookeeper, kafka, extensions, s3, transport);
-
-const healthServer = new HealthProbeServer({
-    bindAddress: healthcheckServer.bindAddress,
-    port: healthcheckServer.port,
-});
+const bucketProcessor = new LifecycleBucketProcessor(
+    zookeeper, kafka, lcConfig, repConfig, s3, transport
+);
 
-function updateBootstrapList() {
-    const { replication } = config.extensions;
-    replication.destination.bootstrapList = config.getBootstrapList();
-
-    config.on('bootstrap-list-update', () => {
-        replication.destination.bootstrapList = config.getBootstrapList();
-    });
+function livenessCheck(res, log) {
+    if (bucketProcessor.isReady()) {
+        sendSuccess(res, log);
+    } else {
+        sendError(res, log, errors.ServiceUnavailable, 'unhealthy');
+    }
 }
 
-function startHealthProbe() {
-    healthServer.onReadyCheck(log => {
-        if (bucketProcessor.isReady()) {
-            return true;
+function probeServerSetup(config, done) {
+    startProbeServer(config, (err, probeServer) => {
+        if (err) {
+            return done(err);
         }
-        log.error('lifecycle bucket processor is not ready!');
-        return false;
-    });
-    logger.info('Starting HealthProbe server');
-    healthServer.start();
-}
 
-function loadManagementDatabase(cb) {
-    return initManagement({
-        serviceName: 'lifecycle',
-        serviceAccount: extensions.lifecycle.auth.account,
-        applyBucketWorkflows: applyBucketLifecycleWorkflows,
-    }, error => {
-        if (error) {
-            logger.error('could not load management db', { error });
-            setTimeout(loadManagementDatabase, 5000);
-            return;
+        if (!probeServer) {
+            logger.info('Skipping lifecycle bucket processor server setup');
+            return done();
         }
-        logger.info('management init done');
-        cb();
+
+        probeServer.addHandler(DEFAULT_LIVE_ROUTE, livenessCheck);
+        probeServer.addHandler(DEFAULT_READY_ROUTE, livenessCheck);
```

Comment:
I have just been using live checks as from what I could tell from the k8s guidelines ready meant something else, but if we should just do the same for both I will update my uses.

Reply:
the bucket/object processor and conductors won't need to handle requests, so there isn't a need for a ready check